ui/app/js/controllers/release_new_controller.js

```diff
@@ -18,6 +18,28 @@ function($scope, $http, $modalInstance, CSRF, Releases, releases) {
   $scope.errors = {};
   $scope.saving = false;
 
+  $scope.fillName = function () {
+    var file = $scope.dataFile;
+
+    var reader = new FileReader();
+    reader.onloadend = function(evt) {
+      var blob = evt.target.result;
+      $scope.$apply( function() {
+         $scope.release.name = JSON.parse(blob).name;
+      });
+    };
+    if (typeof file !== 'undefined') {
+      // should work
+      reader.readAsText(file);
+    }
+
+  };
+
+  $scope.changeName = function () {
+    //wait for actual file to be loaded
+    setTimeout( $scope.fillName , 500);
```

Comment:
Yeah, I've seen that angular doesn't support it that's why I did it like this (it's kinda hacky but it works). I'll try to lower delay but it can't be zero since file won't get cached/loaded or whatever and we will just get file undefined. I can try lowering delay and see how far can I reduce it. It will probably vary a lot depending of internet speed of user, so I picked 500 ms as a safe value initially.